torch/_inductor/lowering.py

```diff
@@ -3027,7 +3027,7 @@ def _new_constant(
         dtype = decode_dtype(dtype) or x.get_dtype()
         device = device or x.get_device()
         size = [sympy.Integer(s) for s in size]
-        return _full(fill_value, device, dtype, size)
+        return _full(fill_value, decode_device(device), dtype, size)
```

Comment:
Why do we have all the decode_device calls ? I would sort of expect this to be normalized already. or maybe we can normalize them in a more central place. it's going to be really easy to forget to do this

Reply:
Yea, i think we should just do this in GraphLowering class prior to invoking the op.. anyway we dont need to block this pr